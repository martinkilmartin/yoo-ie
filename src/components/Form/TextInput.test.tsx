import renderer from 'react-test-renderer'
import TextInput from './TextInput'

test('TextInput displays label and text-input', () => {
  const title = 'Email'
  const placeholder = 'info@yoo.ie'
  const textInputRenderer = renderer.create(
    <TextInput title={title} placeholder={placeholder} />
  )
  const tree = textInputRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const textInputInstance = textInputRenderer.root
  expect(textInputInstance.findByType('span').children[0]).toEqual(title)
  expect(
    textInputInstance.findByProps({
      className: 'input input-primary input-bordered',
    }).props.placeholder
  ).toEqual(placeholder)
})
