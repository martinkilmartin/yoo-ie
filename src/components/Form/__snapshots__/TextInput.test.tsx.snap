// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput displays label and text-input 1`] = `
<div
  className="form-control"
>
  <label
    className="label"
  >
    <span>
      Email
    </span>
     
  </label>
  <input
    className="input input-primary input-bordered"
    maxLength={1000}
    minLength={1}
    placeholder="info@yoo.ie"
    required={true}
    type="text"
  />
</div>
`;
